openlibrary/plugins/admin/code.py

```diff
@@ -389,7 +389,22 @@ def POST_block_account_and_revert(self, account):
             ]
             flattened_records: list[dict] = sum(added_records, [])
             keys_to_delete |= {r['key'] for r in flattened_records}
-            changeset_ids = [c.id for c in changes]
+
+            edition_or_work_keys = {}
+            for change in changes:
+                for item in change.changes:
+                    edition_or_work_keys[item.key] = change.id
```

Comment:
Rename; let's roll with a double for-loop list comprehension bcs I think we tend to prefer list comprehension in the codebase even in this case 

```suggestion
            keys_to_revert = {
                item.key: change.id
                for change in changes
                for item in change.changes
            }
```

Reply:
Ooooh we just found a potential bug here! A user can have multiple changes for the same `item.key` ; so we need this to be a mapping of work key -> a _list_ of change ids, so that we revert all their changes.